### PR TITLE
Add a generic Method for CSV Design UUId

### PR DESCRIPTION
**Pull request by solemabrothers**
The ticket is here: https://app.asana.com/0/1133167201254915/1163465995915835

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java [hunk 1]
```diff
@@ -96,14 +97,20 @@ public String getDescription() {
     @Override
 
     public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
-        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "ActiveoncareList.xls");
+        ReportDesign rd = createExcelTemplateDesign("c2ef160f-f855-462d-a2ad-c38ca3ceffa0", reportDefinition, "ActiveoncareList.xls");
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java:100]: @solemabrothers Why the change, this seems contrary to other approaches

Comment (solemabrothers) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java:100]: For all reports where we have a CSV export, we need to use a different uuid as using the same Excel UUID throws the duplicate uuid entry error.

Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java:100]: @solemabrothers why not add a getCSVDesignUuid() method?

Comment (ssmusoke) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java:100]: @solemabrothers how about the getUuid method

**Comment by ebichete** [comment]
Is there a reason why this construction:

`
public String getCSVDesignUuid()
{
	return "8d711174-7454-4094-a560-00ab51f50ac9";
}
`

is not written as:
`
public static final String CSV_DESIGN_UUID = "8d711174-7454-4094-a560-00ab51f50ac9";
`

Things like `getVersion()` are overriding framework-provided methods from base classes, but otherwise constants should be properly.

**Comment by ssmusoke** [comment]
@ebichete The additional static final is not really needed, since each class is used as a component and never statically called. 

Also the getCSVDesignUuid() is named that way to describe intent and be consistent with the rest of the framework specific calls (but its not enforced)

**Comment by ebichete** [comment]
> @ebichete The additional static final is not really needed, since each class is used as a component and never statically called.
> 
> Also the getCSVDesignUuid() is named that way to describe intent and be consistent with the rest of the framework specific calls (but its not enforced)

Ah, I see. If it is intended as a component interface, then add the definition for `getCSVDesignUuid()` to `UgandaEMRDataExportManager` or an ancestor base class. Like is done with `getExcelDesignUuid()`. And preface the usage here with an `@Override` annotation.

**Comment by solemabrothers** [comment]
> 
> 
> > @ebichete The additional static final is not really needed, since each class is used as a component and never statically called.
> > Also the getCSVDesignUuid() is named that way to describe intent and be consistent with the rest of the framework specific calls (but its not enforced)
> 
> Ah, I see. If it is intended as a component interface, then add the definition for `getCSVDesignUuid()` to `UgandaEMRDataExportManager` or an ancestor base class. Like is done with `getExcelDesignUuid()`. And preface the usage here with an `@Override` annotation.

Noted

**Comment by solemabrothers** [comment]
@ebichete also it will be required that the getCSVDesignUuid() method be implemented by all subclasses of  UgandaEMRDataExportManager. Something we dont need

**Comment by ebichete** [comment]
@solemabrothers If it is only used by a few reports and would affect implementation of the other reports, then create a common base class from `UgandaEMRDataExportManager` for those five reports.

As a rule of thumb, sharing functionality by copy and paste, should not exceed 3 usages.

**Comment by semujju** [comment]
> > > @ebichete The additional static final is not really needed, since each class is used as a component and never statically called.
> > > Also the getCSVDesignUuid() is named that way to describe intent and be consistent with the rest of the framework specific calls (but its not enforced)
> > 
> > 
> > Ah, I see. If it is intended as a component interface, then add the definition for `getCSVDesignUuid()` to `UgandaEMRDataExportManager` or an ancestor base class. Like is done with `getExcelDesignUuid()`. And preface the usage here with an `@Override` annotation.
> 
> @ebichete this is Noted, however, why do we need to define it in the base class, given that it will only be used by a few reports unlike the `getExcelDesignUuid()` which is used by all the reports. I thought it would be better to define it only in classess that will use it.
